### DISCUSSION TITLE
Port `prettier-plugin-sort-imports` to Biome

### DISCUSSION THREAD

**Discussion by steelbrain** [biomejs/biome#979] [RFC]
Very excited about Biome winning the prettier challenge. I tested Biome on a few projects I'm working on and the results are incredible! I'm seeing on the order of 30x improvements, with both linting and formatting, compared to just formatting in Prettier.

I would love to migrate said projects to Biome but I'll be leaving behind a pretty important feature that engineers use daily, which is organizing the imports into groups based on user defined rules, implemented in https://github.com/trivago/prettier-plugin-sort-imports.

The plugin has 2.5k stars on Github, and half a million downloads a week, at the time of writing. Porting said plugin would allow even more people to use Biome and benefit from the incredible speed improvements.

**Comment by ematipico** [comment]
Hey @steelbrain , thank you for considering Biome.

Import sorting is a soft subject in the JS ecosystem, and there are many solutions for achieving that. While trivago created their plugin with prettier, some users use eslint to do so. Even with another plugin https://www.npmjs.com/package/eslint-plugin-simple-import-sort.

Our import sorting is not as mature as the said plugins, and we want to make sure that we **first** fix possible issues, before we can consider additional options. However, we should carefully choose the options because we don't want to end up with issues like [Prettier did](https://prettier.io/docs/en/option-philosophy.html). Also, that's part of our [Technical philosophy](https://biomejs.dev/internals/philosophy#technical). 

Some users have already opened some discussions in this regard: https://github.com/biomejs/biome/discussions/645

**Comment by Netail** [comment]
A refined importer sorter been implemented in Biome V2 :)
